runtime/lua/vim/_editor.lua

```diff
@@ -1171,12 +1169,10 @@ function vim.deprecate(name, alternative, version, plugin, backtrace)
     local displayed = vim._truncated_echo_once(msg)
     return displayed and msg or nil
   else
-    vim.validate {
-      name = { name, 'string' },
-      alternative = { alternative, 'string', true },
-      version = { version, 'string', true },
-      plugin = { plugin, 'string', true },
-    }
+    vim.validate('name', name, 'string')
```

Comment:
Is there a written convention around separating the validation of each field into different `vim.validate` calls?

Reply:
And if there is, it would be nice for it to be documented :)

Reply:
`:help vim.validate` contains:

```vimhelp
    This function has two valid forms:
    1. vim.validate(name: str, value: any, type: string, optional?: bool)
    2. vim.validate(spec: table)

    Form 1 validates that argument {name} with value {value} has the type
    {type}. {type} must be a value returned by |lua-type()|. If {optional} is
    true, then {value} may be null. This form is significantly faster and
    should be preferred for simple cases.
```

Reply:
So the answer is that we haven’t documented in our maintenance guidelines about using the 2nd form.

Reply:
Do we need to document it in maintenance guidelines if it's already documented in the main documentation?

Reply:
Oh nvm, I understand now that the last sentence refers to the use of the function and not the `optional` property. Ignore me then.